application/src/main/java/org/togetherjava/tjbot/commands/moderation/ModerationUtils.java

```diff
@@ -376,33 +377,37 @@ record TemporaryData(Instant expiresAt, String duration) {
+     * @param temporaryData is the message that is being used to display action duration.
+     * @param guild to retrieve the guild.
+     * @param reason for the action.
      * @return the appropriate advice.
      */
-    public static String getDmAdvice(ModerationAction action, String... message) {
-        if (message.length == 3) {
+    public static String getDmAdvice(ModerationAction action,
+            @Nullable ModerationUtils.TemporaryData temporaryData, Guild guild, String reason) {
+        if (action == ModerationAction.MUTE) {
+            String durationMessage =
+                    temporaryData == null ? "permanently" : "for " + temporaryData.duration();
             return """
-                    Hey there, sorry to tell you but unfortunately you have been %s %s from the server %s.
+                    Hey there, sorry to tell you but unfortunately you have been %s %s in the server %s.
                     To get in touch with a moderator, you can simply use the **/modmail** command here in this chat. Your message will then be forwarded and a moderator will get back to you soon :thumbsup:
                     The reason for being %s is: %s
                     """
-                .formatted(action.getVerb(), message[0], message[1], action.getVerb(), message[2]);
-        } else if (ModerationAction.UNMUTE.getVerb().equals(action.getVerb())
-                || ModerationAction.UNQUARANTINE.getVerb().equals(action.getVerb())) {
+                .formatted(action.getVerb(), durationMessage, guild.getName(), action.getVerb(),
+                        reason);
+        } else if (action == ModerationAction.UNMUTE || action == ModerationAction.UNQUARANTINE) {
             return """
                     Hey there, you have been %s in the server %s.
                     This means you can now interact with others in the server again.
                     The reason for being %s is: %s
-                    """.formatted(action.getVerb(), message[0], action.getVerb(), message[1]);
+                    """.formatted(action.getVerb(), guild.getName(), action.getVerb(), reason);
         }
         return """
-                Hey there, sorry to tell you but unfortunately you have been %s, which came from the server %s.
-                To get in touch with a moderator, you can simply use the **/modmail** command here in this chat. Your message will then be forwarded and a moderator will get back to you soon :thumbsup:
+                Hey there, sorry to tell you but unfortunately you have been %s, in the server %s.
+                To get in touch with a moderator, you can simply use the %s command here in this chat. Your message will then be forwarded and a moderator will get back to you soon :thumbsup:
                 The reason for being %s is: %s
                 """
-            .formatted(action.getVerb(), message[0], action.getVerb(), message[1]);
+            .formatted(action.getVerb(), guild.getName(),
```

Comment:
now, normally i would recommend to construct this whole message step by step, based on the arguments. instead of having "if action x -> use this message". but, while it makes the code itself better, it would harm readability of the message. and realistically speaking, we want to be able to easily understand the message being send and adjust it. so maybe we can keep this construct as **intentional duplication**.

but in that case, we should at least do something about this method being so fragile. just imagine someone adding a new temporary mod action and calling this method with temporary data. but then being confused why the message doesnt contain the text about the temporary data... this is bad.

this method must have an explicitly defined list of supported actions and throw an exception if its used with any different exception. only then, people understand how to use this method and what to adjust to add support for their message.

so im suggesting to add constants
```java
private static final Set<ModerationAction> TEMPORARY_ACTIONS = EnumSet.of(MUTE);
private static final Set<ModerationAction> REVOKE_ACTIONS = EnumSet.of(UNMUTE, UNQUARANTINE);
private static final Set<ModerationAction> SOFT_ACTIONS = EnumSet.of(WARN);
```
in this class. and then having a if-else in the method, which throws `IllegalArgumentException` if its unsupported.

now, case 1 and 3 are effectively the same. their only difference is what that one `%s` is. its either:
* empty string if `SOFT_ACTION`
* `permanent` if `TEMPORARY_ACTION` but no `temporaryData` given (`null`)
* `for duration...` if `TEMPORARY_ACTION` with `temporaryData` given

so this should go into the if-else:
```java
if (REVOKE_ACTIONS.contains(action)) {
  return ...
}

String durationMessage;
if (SOFT_ACTION.contains(action)) {
  durationMessage = "";
} else if (TEMPORARY_ACTIONS.contains(action)) {
  durationMessage = temporaryData == null ? "permanently" : "for " + temporaryData.duration();
} else {
  throw new IllegalArgumentException("Action '%s' is not supported by this method".formatted(action));
}

return ...
```